src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsUpdateMetadataStep.java

```diff
@@ -0,0 +1,26 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class ConvertToPredictableFileIdsUpdateMetadataStep extends DefaultUndoStep {
+
+  private final UUID datasetId;
+  private final DatasetService datasetService;
+
+  public ConvertToPredictableFileIdsUpdateMetadataStep(
+      UUID datasetId, DatasetService datasetService) {
+    this.datasetId = datasetId;
+    this.datasetService = datasetService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    datasetService.setPredictableFileIds(datasetId, true);
+    return StepResult.getStepResultSuccess();
+  }
```

Comment:
Should there be an undo step here?

Reply:
We don't need to worry about it since it never gets committed on failure

Reply:
Oh sorry, I this isn't bigquery, this is for Postgres. I didn't want to set it to false automatically (since it could have been true at the start).  Though as I'm typing, I'm realizing that we can just track it at the beginning.  I'll make that change